### PR TITLE
CVQ2-140 Option to initialise compliance tool with a matching dataset

### PR DESCRIPTION
**Pull request by Rosa-Fox**
Co-authored-by: Alan Carter <alan.carter@digital.cabinet-office.gov.uk>

### CODE REVIEW

**Comment by alan-gds** [comment]
After this PR has been merged, https://github.com/alphagov/verify-service-provider/pull/99 can be merged.

If any changes are made as a result of feedback on this PR, CVQ2-141-specify-pid-in-json must be rebased.

File: scripts/init-compliance-tool-for-environment.sh [hunk 1]
```diff
@@ -1,57 +1,85 @@
...
+fi
 
 case $1 in
   	'local')
 		echo "configuring compliance tool for Relying Party running on localhost:3200"
 		SERVICE_ENTITY_ID="http://verify-service-provider-local"
 		ASSERTION_CONSUMER_SERVICE_URL="http://localhost:3200/verify/response"
+		COMPLIANCE_TOOL_INIT_URL="http://localhost:50270/$COMPLIANCE_TOOL_INIT_URL_SLUG"
+    MATCHING_SERVICE_ENTITY_ID="http://www.test-rp-ms.gov.uk/SAML2/MD"
+    MATCHING_SERVICE_SIGNING_KEY=`cat ../test-keys-and-certs/msa-signing-key.pem`
```
Comment (karlbaker02) [inline scripts/init-compliance-tool-for-environment.sh]: Should we be using the same `MATCHING_SERVICE_SIGNING_KEY` specified for the `dev` environment or do we need to add this key to the `test-keys-and-certs` directory? I seem to remember there was some discussion around this.

File: scripts/init-compliance-tool-for-environment.sh [hunk 2]
```diff
@@ -1,57 +1,85 @@
 #!/usr/bin/env bash
 set -e
 
-if [ ! $# -eq 2 ] ; then
-	echo "this script requires the environment and a persistentId to be passed in as arguments. e.g."
-	echo "$ ./init-compliance-tool.sh <environment> <persistentId>"
+if [ ! $# -eq 3 ] ; then
+	echo "this script requires the environment and a persistentId to be passed in as arguments, along with either the '--matching' flag or the name of a file containing matching dataset json. e.g."
+	echo "$ ./init-compliance-tool.sh <environment> <persistentId> --matching"
```
Comment (karlbaker02) [inline scripts/init-compliance-tool-for-environment.sh]: Should we have an additional line of example showing passing in a file, for completeness?

Comment (karlbaker02) [reply scripts/init-compliance-tool-for-environment.sh]: Disregard this, I've started to look at the next PR and have seen the examples there.

File: scripts/init-compliance-tool-for-environment.sh (hunk 1 above)
Comment (Rosa-Fox) [reply scripts/init-compliance-tool-for-environment.sh]: @karlbaker02 we will change this to use the same `MATCHING_SERVICE_SIGNING_KEY` as the dev environment. This eliminates the need for generating that `msa-signing-key.pem` file.

**Pull request merged**